src/defu.ts

```diff
@@ -89,4 +94,16 @@ export const defuArrayFn = createDefu((object, key, currentValue) => {
   }
 });
 
+// Custom version for skipping keys not present in defaults
+export const defuSchema = createDefu((schema, key) => {
+  // Checking for hasOwnProperty is required for compatibility with older browsers
+  if (
+    !Object.hasOwn?.(schema, key) ||
+    !Object.prototype.hasOwnProperty.call(schema, key)
```

Comment:
It is kinda strange because at the same time, we are using the optional chaining syntax (which is certainly not supported in older browsers) but also `Object.prototype.hasOwnProperty.call`. I think it would be the bundler's job to transform to the older syntax when it is configured to and instead we can keep the syntax clean and simple.

Reply:
We could: 1) Avoid optional chaining syntax (see my more longer comment https://github.com/unjs/defu/pull/112#issuecomment-1802669657) 2) directly check property existence using `in` keyword and let the bundlers do necessary transform for backward.

Reply:
@pi0 Good catch, don't know what I was thinking! 😅